Sources/ClientRuntime/Networking/Http/Middlewares/FlexibleChecksumsRequestMiddleware.swift

```diff
@@ -0,0 +1,86 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0.
+
+public struct FlexibleChecksumsRequestMiddleware<OperationStackInput, OperationStackOutput>: Middleware {
+
+    public let id: String = "FlexibleChecksumsRequestMiddleware"
+
+    let checksumAlgorithms: [String]
+
+    public init(checksumAlgorithms: [String]) {
+        self.checksumAlgorithms = checksumAlgorithms
+    }
+
+    public func handle<H>(context: Context,
+                          input: SerializeStepInput<OperationStackInput>,
+                          next: H) async throws -> OperationOutput<OperationStackOutput>
+    where H: Handler,
+    Self.MInput == H.Input,
+    Self.MOutput == H.Output,
+    Self.Context == H.Context {
+        
+        // Initialize logger
+        guard let logger = context.getLogger() else {
+            return try await next.handle(context: context, input: input)
```

Comment:
Q: So if a logger wasn't set on the middleware context, we skip the flexible checksum request middleware? What's the reason?

Reply:
Good catch I meant to change this as I did in response to `throw ClientError.unknownError("No logger found!")`